src/subsystems/rust/translators/cargo-lock/default.nix

```diff
@@ -97,17 +97,58 @@ in {
     parsedLock = projectTree.files."Cargo.lock".tomlContent;
     parsedDeps = parsedLock.package;
 
+    makeDepId = dep: "${dep.name} ${dep.version} (${dep.source or ""})";
+
     # Gets a checksum from the [metadata] table of the lockfile
-    getChecksum = dep: let
-      key = "checksum ${dep.name} ${dep.version} (${dep.source})";
-    in
-      parsedLock.metadata."${key}";
+    getChecksum = dep: parsedLock.metadata."checksum ${makeDepId dep}";
 
     # map of dependency names to versions
     depNamesToVersions =
-      l.listToAttrs
-      (l.map (dep: l.nameValuePair dep.name dep.version) parsedDeps);
-    # This parses a "package-name version" entry in the "dependencies"
+      l.foldl'
+      (
+        all: el:
+          if l.hasAttr el.name all
+          then all // {${el.name} = all.${el.name} ++ [el];}
+          else all // {${el.name} = [el];}
+      )
+      {}
+      parsedDeps;
+    findUniqDep = let
+      # tests whether if el contains all elements from dep
```

Comment:
What is `el` and what kind of `elements` are we talking about here? Maybe give more meaningful names?

Reply:
Done, added more comments